From 669f9ca2ee3bcefa5fbda9af20e2968e28e70c53 Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@caviumnetworks.com>
Date: Thu, 29 Nov 2018 11:42:19 -0800
Subject: [PATCH 0829/1051] mmc: octeontx2: modify check for 8xxx/9xxx chips

Use only most significant 12..15 bits for finding the
type of chip.

Change-Id: I4b723cfd7af2639bb1bb60bf160604f35250acf9
Signed-off-by: Sujeet Baranwal <sbaranwal@caviumnetworks.com>
Reviewed-on: https://sj1git1.cavium.com/1143
Tested-by: Nadav Haklai <Nadav.Haklai@cavium.com>
Reviewed-by: Sujeet Kumar Baranwal <Sujeet.Baranwal@cavium.com>
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/mmc/host/cavium.h | 19 ++++++++-----------
 1 file changed, 8 insertions(+), 11 deletions(-)

diff --git a/drivers/mmc/host/cavium.h b/drivers/mmc/host/cavium.h
index 47f39e876ada..4b490beec38f 100644
--- a/drivers/mmc/host/cavium.h
+++ b/drivers/mmc/host/cavium.h
@@ -24,11 +24,8 @@
 #define CAVIUM_MAX_MMC		4
 
 /* Subsystem Device ID */
-#define PCI_SUBSYS_DEVID_88XX   0xA100
-#define PCI_SUBSYS_DEVID_81XX   0xA200
-#define PCI_SUBSYS_DEVID_83XX   0xA300
-#define PCI_SUBSYS_DEVID_96XX	0xB200
-#define PCI_SUBSYS_DEVID_95XX   0xB300
+#define PCI_SUBSYS_DEVID_8XXX	0xA
+#define PCI_SUBSYS_DEVID_9XXX	0xB
 
 #define KHZ_400 (400000)
 #define MHZ_26  (26000000)
@@ -282,25 +279,25 @@ extern const char *cvm_mmc_irq_names[];
 static inline bool is_mmc_8xxx(struct cvm_mmc_host *host)
 {
 	struct pci_dev *pdev = host->pdev;
+	u32 chip_id = (pdev->subsystem_device >> 12) & 0xF;
 
-	return (pdev->subsystem_device == PCI_SUBSYS_DEVID_81XX) ||
-		(pdev->subsystem_device == PCI_SUBSYS_DEVID_83XX) ||
-		(pdev->subsystem_device == PCI_SUBSYS_DEVID_88XX);
+	return (chip_id == PCI_SUBSYS_DEVID_8XXX);
 }
 
 static inline bool is_mmc_otx2(struct cvm_mmc_host *host)
 {
 	struct pci_dev *pdev = host->pdev;
+	u32 chip_id = (pdev->subsystem_device >> 12) & 0xF;
 
-	return (pdev->subsystem_device == PCI_SUBSYS_DEVID_96XX) ||
-		(pdev->subsystem_device == PCI_SUBSYS_DEVID_95XX);
+	return (chip_id == PCI_SUBSYS_DEVID_9XXX);
 }
 
 static inline bool is_mmc_otx2_A0(struct cvm_mmc_host *host)
 {
 	struct pci_dev *pdev = host->pdev;
+	u32 chip_id = (pdev->subsystem_device >> 12) & 0xF;
 
 	return (pdev->revision == 0x00) &&
-		(pdev->subsystem_device == PCI_SUBSYS_DEVID_96XX);
+		(chip_id == PCI_SUBSYS_DEVID_9XXX);
 }
 #endif
-- 
2.17.1

